File: BE/ClinicBooking/Models/Appointment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ClinicBooking.Models
{
    public class Appointment : EntityBase
    {
        [Key]
        public int AppointmentID { get; set; }
        [Required]
        [ForeignKey("Doctor")]
        public int DoctorId { get; set; }
        public Doctor Doctor { get; set; }
        [Required]
        [ForeignKey("BookByUser")]
        public int BookByUserID { get; set; }
        public User BookByUser { get; set; }
        public DateTime StartTime { get; set; }
        [Required]
        public DateTime EndTime { get; set; }
        public decimal? Price { get; set; }
        [Required]
        public AppointmentStatus AppointmentStatus { get; set; }
        public int? MedicalHistoryId { get; set; }
        public MedicalHistory MedicalHistory { get; set; }
    }

    public enum AppointmentStatus
    {
        Booked,
        Pending,
        Examined
    }
}

File: fe/types/appointment.ts
 
enum AppointmentStatus {
  Booked = 0,
  Pending = 1,
  Examined = 2,
}

export type Appointment = {
  id: number;
  doctorId: number;
  bookByUserId: number;
  startTime: Date | string;
  endTime: Date | string;
  price: number;
  appointmentStatus:
    | AppointmentStatus.Booked
    | AppointmentStatus.Pending
    | AppointmentStatus.Examined;
  medicalHistoryId: number | null;
  active: boolean;
};

export interface Pagination {
  pageNumber: number;
  pageSize: number;
  totalItems: number;
  totalPages: number;
}

export interface AppointmentsResponse {
  status: number;
  message: string;
  data: Appointment[];
  pagination: Pagination;
}

export interface AppointmentResponse {
  status: number;
  message: string;
  data: Appointment;
}

export interface AppointmentParams {
  pageSize?: number;
  pageNumber?: number;
}
